### PR TITLE
ci: add spellcheck to ci tests

### LINKED ISSUE #255: Add spellcheck to ci tests
Add spellcheck to ci tests, this will ensure any spelling errors or typos are found.
We can generate a dictionary to add ignored words.

### PR DESCRIPTION
**Pull request by davemooreuws**
Closes #255

### CODE REVIEW

**Review by asalkeld — APPROVED** [review]
lgtm

File: dictionary.txt [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+unhandled
+queryable
+composable
+integrations
+Scalable
+Transactional
+Pulumi
+homebrew
+PubSubPull
+\..+
```
Comment (tjholm) [inline dictionary.txt:168]: can we use regex's in some of these cases?
e.g. br and br/ and other markup could be ignored using a standard html tag regex?

Comment (davemooreuws) [reply dictionary.txt:168]: Yes we can, this dictionary was auto generated by the library.

Comment (davemooreuws) [reply dictionary.txt:168]: @tjholm this is more for unknown words in the en-US dictionary. You can use regex for words, like the one for any with a dot in front of it.

**Review by raksiv — CHANGES_REQUESTED** [review]
What locale is this going against? Looks like AU - e.g. behavior is spelt correctly per US dictionary.

**Comment by davemooreuws** [comment]
> What locale is this going against? Looks like AU - e.g. behavior is spelt correctly per US dictionary.

The config is set to language: en-US maybe it isn't working.

**Review by raksiv — APPROVED** [review]

**Pull request merged**